samples/bluetooth/peripheral_nus/src/main.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/services/nus.h>
+
+#define DEVICE_NAME		CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN		(sizeof(DEVICE_NAME) - 1)
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static const struct bt_data sd[] = {
+	BT_DATA_BYTES(BT_DATA_UUID128_ALL, BT_UUID_NUS_SRV_VAL),
+};
+
+static void notif_enabled(bool enabled, void *ctx)
+{
+	ARG_UNUSED(ctx);
+
+	printk("%s() - %s\n", __func__, (enabled ? "Enabled" : "Disabled"));
+}
+
+static void received(const void *data, uint16_t len, void *ctx)
+{
+	char message[CONFIG_BT_L2CAP_TX_MTU + 1] = "";
+
+	ARG_UNUSED(ctx);
+
+	memcpy(message, data, MIN(sizeof(message) - 1, len));
+	printk("%s() - Len: %d, Message: %s\n", __func__, len, message);
+}
+
+struct bt_nus_cb nus_listener = {
+	.notif_enabled = notif_enabled,
+	.received = received,
+};
+
+int main(void)
+{
+	int err = -1;
```

Comment:
And here. Btw, even if it was justitified to initialize it, using `-1` is not really a good idea. Zephyr generally uses negative posix errno values for `int` type errors, so there's a risk someone might interpret `-1` as `-EPERM`.